spec/domain/command_handlers/game_command_handler_spec.rb

```diff
@@ -116,231 +137,335 @@
-              InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(110)),
-              GameEndedEvent.new(id, GameState::ENDED)
-            ]
-          end
-        end
+      it 'wins the game for a player' do
+        when_command PlayCurrentRoundCommand.new(id: id)
+        then_events(
+          GameRoundStartedPlayingEvent.new(id, GameRound.new(1)),
+          InstructionCardRevealedEvent.new(id, bob, InstructionCard.u_turn(10)),
+          RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
+          InstructionCardRevealedEvent.new(id, bob, InstructionCard.u_turn(20)),
+          RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::DOWN)),
+          InstructionCardRevealedEvent.new(id, bob, InstructionCard.u_turn(30)),
+          RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
+          InstructionCardRevealedEvent.new(id, bob, InstructionCard.u_turn(40)),
+          RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::DOWN)),
+          InstructionCardRevealedEvent.new(id, bob, InstructionCard.u_turn(50)),
+          RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
+          GoalTouchedEvent.new(id, bob, Goal.new(0, 0, 1)),
+          PlayerWonGameEvent.new(id, bob),
+          SpawnReplacedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(10)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(20)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(30)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(40)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(50)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(60)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(70)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(90)),
+          InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(110)),
+          GameRoundFinishedPlayingEvent.new(id, GameRound.new(1)),
+          GameEndedEvent.new(id, GameState::ENDED)
+        )
       end
     end
   end
 
   describe JoinGameCommand do
-    let(:command) { join_game_command }
-
-    it_behaves_like "an event publisher" do
-      before { dispatch(create_game_command) }
+    subject(:command) { JoinGameCommand.new(id: id, player_id: steven) }
+    before { given_events GameCreatedEvent.new(id, GameState::LOBBYING, bob) }
 
-      let(:expected_events) { [PlayerJoinedGameEvent.new(id, steven)] }
+    it 'joins a game' do
+      when_command(command)
+      then_events(PlayerJoinedGameEvent.new(id, steven))
     end
 
-    context "when already in game" do
-      before do
-        dispatch(create_game_command)
-        dispatch(join_game_command)
-      end
+    context 'given the player already joined the game' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.